### DISCUSSION TITLE
GraphQL API development mode

### DISCUSSION THREAD

**Discussion by letmejustputthishere** [community/community#24612] [Apps, API and Webhooks]
<p>I’m going to code an artifact for my thesis that heavily relies on GitHubs GraphQL API. The docs say the following:</p>
<blockquote>
<p>The GraphQL API v4 rate limit is <strong>5,000 points per hour</strong> .</p>
</blockquote>
<p>Is there a way to remove this limit when developing locally? Or is there some sort of “research” plan one can subscribe to to have more points per hour?</p>

**Comment by letmejustputthishere** [comment]
<p>bump and 20 chars so i can bump</p>

**Comment by letmejustputthishere** [comment]
<p>i’m bumping again in hope of getting an answer</p>

**Comment by queenofcorgis** [comment, marked as answer]
<p>Hey there <a class="mention" href="/letmejustputthishere">@letmejustputthishere</a> :wave:t3: This isn’t my area of expertise, but I know you’ve been waiting a bit for an answer so let me try and help.</p>
<p>GitHub’s users can’t increase primary API rate limits directly since that would defeat the purpose of having limits. However, there are a couple workarounds to get a higher primary API rate limit:</p>
<ul>
<li>Upgrading to a GHEC plan.
<ul>
<li>GHEC organizations have increased limits for OAuth Apps and GitHub Apps.</li>
</ul>
</li>
<li>Using a GitHub App with an organization.
<ul>
<li>GitHub Apps have dynamic rate limits which grow with the size of the organization, e.g. with the number of repositories.</li>
</ul>
</li>
</ul>
